### ISSUE TITLE
win16.h re-defines NULL in a way that conflicts with stdio.h and stdlib.h

### ISSUE THREAD

**Issue by joncampbell123** [open-watcom/open-watcom-v2#378]
While compiling DOSLIB hw/dos:

    /usr/src/open-watcom-v2/rel/h/stdio.h(139): Error! E1100: Definition of macro 'NULL' not identical to previous definition
    /usr/src/open-watcom-v2/rel/h/stdio.h(139): Note! N2002: 'NULL' defined in: /usr/src/open-watcom-v2/rel/h/win/win16.h(131)
    /usr/src/open-watcom-v2/rel/h/stdlib.h(112): Error! E1100: Definition of macro 'NULL' not identical to previous definition
    /usr/src/open-watcom-v2/rel/h/stdlib.h(112): Note! N2002: 'NULL' defined in: /usr/src/open-watcom-v2/rel/h/win/win16.h(131)
    /usr/src/open-watcom-v2/rel/h/string.h(112): Error! E1147: Too many errors: compilation aborted

NULL in stdio.h:

    #define NULL ((void *)0)

NULL in win16.h

    #ifndef NULL
        #define NULL        0
    #endif

**Comment by joncampbell123** [comment]
This is in a source file that includes windows.h first, followed by stdio.h.

From hw/dos/dos.c

    #ifdef TARGET_WINDOWS
    # include <windows.h>
    #endif

    #include <stdio.h>
    #include <conio.h> /* this is where Open Watcom hides the outp() etc. functions */
    #include <stdlib.h>
    #include <string.h>
    #include <stddef.h>
    #include <unistd.h>
    #include <malloc.h>
    #include <assert.h>
    #include <fcntl.h>
    #include <dos.h>

**Comment by joncampbell123** [comment]
As a temporary measure I've added #ifndef NULL...#endif around #define NULL in those headers in my branch of open watcom v2.

**Comment by jmalak** [comment]
Yes, problem exists now, but it was introduced in OW 1.8 when definition of NULL was changed. Two definition of NULL exists for 16-bit Windows code. One Microsoft in win16.h and second C in standard C header files.
It is visible now because I changed standard C header file to have strict NULL definition without #ifndef NULL guard to be sure the same definition of NULL is used overall C code.
I will return back #ifndef NULL guard to suppress this message, but it doesn't fix source of this problem.
Changing NULL in win16.h is not possible for backward compatibility.
